source/developers/plugins/site-plugins.rst

```diff
@@ -29,11 +29,30 @@ A site plugin can contain one or more extensions for Crafter CMS in a single pac
 .. _how-do-i-make-my-own-site-plugin:
 
 ---------------------------------
-How do I make my own site plugin?
+How Do I Make My Own Site Plugin?
 ---------------------------------
 
-A site plugin consist of a group of files that are copied to the site repository when installed, to create your own
-site plugin you only need to copy the files following this structure:
+^^^^^^^^^^^^
+Requirements
+^^^^^^^^^^^^
+You'll need the following for creating your plugin:
+
+* A plugin descriptor file, ``craftercms-plugin.yaml``
+* Your plugin files
+
+The ``craftercms-plugin.yaml`` file contains information about your plugin, such as what license your plugin supports,
+which versions of Crafter CMS is supported, etc.
+
+See :ref:`craftercms-plugin-yaml-file` for more information on what's inside the plugin descriptor.
+
+Your plugin files/folders could be JavaScript files, XML files, etc. depending on the plugin type you're creating.
```

Comment:
Your plugin files/folders could be JavaScript files, XML files, Groovy scripts, images, CSS files, and more depending on the plugin type you're creating.